lib/Jsrt/ChakraCore.h

```diff
@@ -163,4 +163,346 @@ JsGetModuleHostInfo(
+/// <remarks>
+///     <para>
+///     Property IDs are specific to a context and cannot be used across contexts.
+///     </para>
+///     <para>
+///     Requires an active script context.
+///     </para>
+/// </remarks>
+/// <param name="name">
+///     The name of the property ID to get or create. The name may consist of only digits.
+///     The string is expected to be encoded as utf8.
+/// </param>
+/// <param name="length">length of the name</param>
+/// <param name="propertyId">The property ID in this runtime for the given name.</param>
+/// <returns>
+///     The code <c>JsNoError</c> if the operation succeeded, a failure code otherwise.
+/// </returns>
+CHAKRA_API
+    JsCreatePropertyIdUtf8(
+        _In_z_ const char *name,
+        size_t length,
+        _Out_ JsPropertyIdRef *propertyId);
+
+/// <summary>
+///     Gets the name associated with the property ID.
+/// </summary>
+/// <remarks>
+///     <para>
+///     Requires an active script context.
+///     </para>
+///     <para>
+///     Consumer is responsible from calling JsStringFree to free allocated memory.
+///
+///     Experimental. We may update the name or behavior until it is stable.
+///     </para>
+/// </remarks>
+/// <param name="propertyId">The property ID to get the name of.</param>
+/// <param name="buffer">The name associated with the property ID, encoded as utf8.</param>
+/// <param name="bufferSize">Size of the buffer.</param>
+/// <param name="written">Total number of characters written or to be written</param>
+/// <returns>
+///     The code <c>JsNoError</c> if the operation succeeded, a failure code otherwise.
+/// </returns>
+CHAKRA_API
+    JsWritePropertyIdUtf8(
+        JsPropertyIdRef propertyId,
+        uint8_t* buffer,
+        size_t bufferSize,
+        size_t* length);
+
+/// <summary>
+///     Serializes a parsed script to a buffer than can be reused.
+/// </summary>
+/// <remarks>
+///     <para>
+///     <c>JsSerializeScript</c> parses a script and then stores the parsed form of the script in a
+///     runtime-independent format. The serialized script then can be deserialized in any
+///     runtime without requiring the script to be re-parsed.
+///     </para>
+///     <para>
+///     Requires an active script context.
+///     </para>
+/// </remarks>
+/// <param name="script">The script to serialize</param>
+/// <param name="buffer">The buffer to put the serialized script into. Can be null.</param>
+/// <param name="bufferSize">
+///     On entry, the size of the buffer, in bytes; on exit, the size of the buffer, in bytes,
+///     required to hold the serialized script.
+/// </param>
+/// <param name="parseAttributes">Encoding for the script.</param>
+/// <returns>
+///     The code <c>JsNoError</c> if the operation succeeded, a failure code otherwise.
+/// </returns>
+CHAKRA_API
+    JsSerialize(
+        JsValueRef script,
+        BYTE *buffer,
+        unsigned int *bufferSize,
+        JsParseScriptAttributes parseAttributes);
```

Comment:
Suggest putting the parseAttributesinput before the outputs (buffer and buffersize)